poetry/publishing/uploader.py

```diff
@@ -265,13 +265,22 @@ def _upload_file(
                         allow_redirects=False,
                         headers={"Content-Type": monitor.content_type},
                     )
-                if dry_run or resp.ok:
+                if dry_run or resp.status_code == 200:
```

Comment:
As was mentioned on the issue; not all indices return 200 on suceess. And hence why it was recommended to use ok and not 30x. Alternatively accept the 200 range.